### ISSUE TITLE
[Question] Why can't the first line of a multi-line parameters declaration in a function call be an assignement ?

### ISSUE THREAD

**Issue by johnchristopher** [WordPress-Coding-Standards/WordPress-Coding-Standards#1462]
Hello,

PHPCS will trigger a `Assignments must be the first block of code on a line` warning when using aligned named parameters in a function call:
```php
add_submenu_page(
	$parent_slug = 'edit.php?post_type=' . CPT_NAME,
	$page_title  = 'Test',
	$menu_title  = 'Test',
	$capability  = 'manage_options',
	$menu_slug   = 'my-plugin-test',
	$function    = array( &$test, 'display_page' )
);
```
For clarity purposes I like using named parameters.
The https://make.wordpress.org/core/handbook/best-practices/coding-standards/php/ doesn't say anything about named parameters.
What's the reason the first line of a multi-line parameters declaration can't be an assignment ?

Is it more correct to do something like this if I want a visual reminder of the function parameters ?
```php
$args = array(
	$parent_slug => 'test',
	$page_title  => 'test',
	$menu_title  => 'Test',
	$capability  => 'manage_options',
	$menu_slug   => 'my-plugin-test',
	$function    => array( &$test, 'display_page' ),
);
add_submenu_page( $args[ 'parent_slug' ], $args[ 'page_title' ],  $args[ 'page_title' ],  $args[ 'capability' ],  $args[ 'menu_slug' ],  $args[ 'function' ] );
```

**Comment by jrfnl** [comment]
@johnchristopher Creating temporary variables which are never used, is not a good idea as it will pollute the global (or local) namespace and have a - no matter how small - impact on PHP memory usage and potentially execution time.

Why not use comments instead ? That's what they are for, after all.
```php
add_submenu_page(
	'edit.php?post_type=' . CPT_NAME,
	'Test', // Page title.
	'Test', // Menu title.
	'manage_options', // Capability.
	'my-plugin-test', // Menu slug.
	array( &$test, 'display_page' ) // Call-back function.
);
```

**Comment by johnchristopher** [comment]
> Why not use comments instead ? That's what they are for, after all.
It might be a matter of preferences but I find it easier to read aligned named parameters than comments. Right now I settled on this:
```php
add_submenu_page(
	'edit.php?post_type=' . CPT_NAME, // $parent_slug
	$page_title  = 'Test',
	$menu_title  = 'Test',
	$capability  = 'manage_options',
	$menu_slug   = 'my-plugin-test',
	$function    = array( &$test, 'display_page' )
);
```
It isn't satisfactory though since the rest of the named parameters are visually coherent.

**Comment by jrfnl** [comment]
> It isn't satisfactory though since the rest of the named parameters are visually coherent.

What do you mean by this ?

**Comment by johnchristopher** [comment]
> That still doesn't make it clear to me what you expect us to do about this ... ?

I wonder why this piece of code triggers a warning:
```php
add_submenu_page(
	$parent_slug = 'edit.php?post_type=' . CPT_NAME,
	$page_title  = 'Test',
	$menu_title  = 'Test',
	$capability  = 'manage_options',
	$menu_slug   = 'my-plugin-test',
	$function    = array( &$test, 'display_page' )
}
```
when this one doesn't:
```php
add_submenu_page(
	'edit.php?post_type=' . CPT_NAME,
	$page_title  = 'Test',
	$menu_title  = 'Test',
	$capability  = 'manage_options',
	$menu_slug   = 'my-plugin-test',
	$function    = array( &$test, 'display_page' )
);
```

> For now, I can only re-iterate that what you are doing is bad practice.

How is the first piece of code worse than the second one ?

Granted, named parameters could be a problem down the line but if it's the reason the first named parameter triggers a warning in the first piece of code then it follows that the second piece of code should trigger a warning for the `$page_title` assignment and the following assignments.

**Comment by jrfnl** [comment]
> How is the first piece of code worse than the second one ?

That actually looks like a bug in the sniff to me, but as it's a [PHPCS](https://github.com/squizlabs/PHP_CodeSniffer) native sniff, you'd need to ask there to be sure.

**Comment by jrfnl** [comment]
Closing as there's nothing to do for us here and there's been no response from the op in six months.
